### DISCUSSION TITLE
why the flux is different in GapHeatTransfer

### DISCUSSION THREAD

**Discussion by wangzhaohao** [idaholab/moose#25890] [Q&A Modules: General] [answered]
Hello, I am gong to have a test about the thermal flux between the gap. I find moose have tests about [this](https://github.com/idaholab/moose/blob/next/modules/heat_transfer/test/tests/gap_heat_transfer_htonly/planar_xy.i). It's all fine. But when i use 'GapHeatTransfer` into this object.
![image](https://github.com/idaholab/moose/assets/51054734/b7bf1010-168e-40a1-8257-0e0743538366)
the input is flowing
```Assembly
[Mesh]
  file = flux_test.e
[]

[Variables/T]
  initial_condition = 200
[]

[Kernels]
  [diff]
    type = Diffusion
    variable = T
  []
  [body]
    type = BodyForce
    variable = T
    block = 2
    value = 10
  []
[]

[BCs]
  [fix_bc]
    type = DirichletBC
    variable = T
    boundary = 5
    value = 100
  []
[]

[Postprocessors]
inactive = 'block_1_top block_2_down'
  [block_2_right]
    type = SideDiffusiveFluxIntegral
    variable = T
    boundary = 1
    diffusivity = 1
  []

  [block_1_left]
    type = SideDiffusiveFluxIntegral
    variable = T
    boundary = '2'
    diffusivity = 1
  []

  [block_1_right]
    type = SideDiffusiveFluxIntegral
    variable = T
    boundary = '5'
    diffusivity = 1
  []
  [block_1_left_left]
    type = SideDiffusiveFluxIntegral
    variable = T
    boundary = '7 8'
    diffusivity = 1
  []
  [block_1_top]
    type = SideDiffusiveFluxIntegral
    variable = T
    boundary = 3
    diffusivity = 1
  []

  [block_2_down]
    type = SideDiffusiveFluxIntegral
    variable = T
    boundary = 4
    diffusivity = 1
  []
[]

[ThermalContact]
active = 'thermalcontact_left_right'
  [thermalcontact_up_down]
    type = GapHeatTransfer
    variable = T
    primary = 4
    secondary= 3
    gap_conductivity = 0.5
    quadrature = true
    emissivity_primary = 0.5
    emissivity_secondary = 0.5
  []
  [thermalcontact_left_right]
    type = GapHeatTransfer
    variable = T
    primary = 2
    secondary= 1
    gap_conductivity = 1
    quadrature = true
    emissivity_primary = 0.0
    emissivity_secondary = 0.0
  []
[]

[Executioner]
  type = Transient

  dt = 0.1
  num_steps = 10

  nl_abs_tol = 1e-7
  nl_rel_tol = 1e-12
[]

[Outputs]
  exodus = true
[]

```
the flux is differnent!
```
|   time           | block_1_left   | block_1_left_left | block_1_right  | block_2_right  |
|   0.000000e+00 |   0.000000e+00 |      0.000000e+00 |   0.000000e+00 |   0.000000e+00 |
|   1.000000e-01 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
|   2.000000e-01 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
|   3.000000e-01 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
|   4.000000e-01 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
|   5.000000e-01 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
|   6.000000e-01 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
|   7.000000e-01 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
|   8.000000e-01 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
|   9.000000e-01 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
|   1.000000e+00 |  -2.033598e+00 |     -7.418476e-09 |   2.494809e+00 |   2.187500e+00 |
```
and the temperature
![image](https://github.com/idaholab/moose/assets/51054734/20903a19-fc32-4ea1-8ad5-92eed3f825c1)

there I use comsol to solve same test.
![image](https://github.com/idaholab/moose/assets/51054734/1f13e395-4148-425d-9c51-a07a44be7d66)
and output the flux (use probe)
![image](https://github.com/idaholab/moose/assets/51054734/94400b33-7507-43c6-ae56-19686d1a108f)
and there is my [mesh file](https://github.com/wangzhaohao/PKE/blob/main/flux_test.e)
so why the flux is different?
thanks for you advice.

**Comment by GiudGiud** [comment]
Hello

First you need to make sure your results are converged.
Then you should read the documentation for SideDiffusiveFluxIntegral to make sure this is actually what you want. 
it wont count the radiative flux for example

Guillaume

**Reply by wangzhaohao** [reply]
Both is converge and I have see the code. and I presume there is no rediation. Personally, I think i set there is only `gap_conductivity`. What's wrong in the input about the SideDiffusiveFluxIntegral?

**Reply by GiudGiud** [reply]
SideDiffusiveFluxIntegral is not coded for interfaces for example. As the documentation says the underlying discretization of the flux has to match between your kernels and the postprocessor for it to be accurate

**Comment by wangzhaohao** [comment]
> As the documentation says the underlying discretization of the flux has to match between your kernels and the postprocessor for it to be accurate

 how to understander this? could you tell me use which postprocessors to get the accurate flux? thanks for you help.

**Reply by GiudGiud** [reply, marked as answer]
SideDiffusiveFlux is not written to work with GapHeatTransfer.
You ll need to write one most likely, I dont know that there exists one

Looking back at the entire post you get pretty good agreement on temperature right? it s just the flux that's different?

**Reply by wangzhaohao** [reply]
Yes. you are right. Only the flux is different. the temperature is right. For a moment, I thought there was something wrong with the theory I learned. 🤪Thanks for you advice.
